packages/preview-middleware/src/base/flp.ts

```diff
@@ -0,0 +1,188 @@
+import { json } from 'express';
+
+/**
+ * Default theme
+ */
+const DEFAULT_THEME = 'sap_horizon';
+
+/**
+ * Default path for mounting the local FLP.
+ */
+const DEFAULT_PATH = '/test/flp.html';
+
+/**
+ * Internal structure used to fill the sandbox.html template
+ */
+export interface TemplateConfig {
+    apps: Record<
+        string,
+        {
+            title: string;
+            description: string;
+            additionalInformation: string;
+            applicationType: 'URL';
+            url: string;
+        }
+    >;
+    ui5: {
+        libs: string;
+        theme: string;
+        flex: {
+            applyConnector: string;
+            writeConnector: string;
+            custom: boolean;
+        }[];
+        resources: Record<string, string>;
+    };
+}
+
+/**
+ * Class handling preview of a sandbox FLP.
+ */
+export class FlpSandbox {
+    protected templateConfig: TemplateConfig;
+    protected readonly config: FlpConfig;
+    public readonly router: any;
+
+    /**
+     * Constructor setting defaults and keeping reference to workspace resources.
+     *
+     * @param config configuration from the ui5.yaml
+     * @param project reference to the project provided by the UI5 CLI
+     * @param utils middleware utilities provided by the UI5 CLI
+     * @param logger logger instance
+     */
+    constructor(
+        config: Partial<FlpConfig>,
+        private readonly project: ReaderCollection,
+        private readonly utils: MiddlewareUtils,
+        private readonly logger: Logger
+    ) {
+        this.config = {
+            path: config.path ?? DEFAULT_PATH,
+            apps: config.apps ?? []
+        };
+        logger.debug(`Config: ${JSON.stringify(this.config)}`);
+        this.router = createRouter();
+    }
+
+    /**
+     * Initialize the FLP sandbox router.
+     *
+     * @param manifest application manifest
+     */
+    init(manifest: Manifest): void {
+        const flex = this.createFlexHandler();
+        const supportedThemes: string[] = (manifest['sap.ui5']?.supportedThemes as []) ?? [DEFAULT_THEME];
+        this.templateConfig = {
+            apps: {},
+            ui5: {
+                libs: Object.keys(manifest['sap.ui5']?.dependencies?.libs ?? {}).join(','),
+                theme: supportedThemes.includes(DEFAULT_THEME) ? DEFAULT_THEME : supportedThemes[0],
+                flex,
+                resources: {}
+            }
+        };
+        this.addApp(manifest, {
+            target: '/',
+            local: '.',
+            intent: {
+                object: 'app',
+                action: 'preview'
+            }
+        });
+
+        // add route for the sandbox.html
+        this.router.get(this.config.path, (_req: Request, res: Response) => {
+            const template = readFileSync(join(__dirname, '../../templates/flp/sandbox.html'), 'utf-8');
+            res.status(200);
+            res.send(render(template, this.templateConfig));
```

Comment:
Need to rewrite to

```
 if (res['_livereload']) {
        res.write(html);
        res.end();
    } else {
        res.status(200).contentType('html').send(html);
    }
```
in order to add the livereload feature (if available) to the generated html file

Reply:
https://github.com/SAP/open-ux-tools/pull/1093/commits/10ddec591f6408852e89eda14305185f936798f5

Reply:
thx